third_party/nvfuser/csrc/lower_utils.cpp

```diff
@@ -745,6 +745,28 @@ bool isScalarExpr(Expr* expr) {
   return true;
 }
 
+bool dependsOnThreadNamedScalars(Val* val) {
+  if (val == nullptr) {
+    // Handle trivial case.
+    return false;
+  }
+
+  auto inputs = InputsOf::output(val->fusion(), val);
+
+  for (auto input : inputs) {
+    if (auto ns = dynamic_cast<NamedScalar*>(input)) {
+      for (auto ptype : kParallelTypeTIDs) {
+        if (ns->name() == stringifyThread(ptype) ||
+            ns->name() == stringifyThreadSize(ptype)) {
```

Comment:
Do thread sizes matter?

Reply:
No, I removed this function completely, and uses `getRegisterType` from expr simplifier to replace it.